src/api/api.c

```diff
@@ -219,12 +219,45 @@ int8_t zp_scouting_config_insert(z_loaned_scouting_config_t *sc, uint8_t key, co
     return _zp_config_insert(sc, key, value);
 }
 
-z_encoding_t z_encoding(z_encoding_prefix_t prefix, const char *suffix) {
-    return (_z_encoding_t){
-        .id = prefix, .schema = _z_bytes_wrap((const uint8_t *)suffix, (suffix == NULL) ? (size_t)0 : strlen(suffix))};
+// TODO: STRING BASED ENCODING FEATURE
+int8_t zp_encoding_make(z_owned_encoding_t *encoding, z_encoding_id_t id, const char *schema) {
+    // Init encoding
+    encoding->_val = (_z_encoding_t *)z_malloc(sizeof(_z_encoding_t));
+    if (encoding->_val == NULL) {
+        return _Z_ERR_SYSTEM_OUT_OF_MEMORY;
+    }
+    encoding->_val->id = id;
+    // Clone schema
+    if (schema != NULL) {
```

Comment:
Will be better call `_z_encoding_make` (but it should be changed to copy prefix, current version only wrap it)

Reply:
The copy version is used internally, so I'll need the two version.